packages/askar/src/wallet/AskarProfileWallet.ts

```diff
@@ -89,19 +86,15 @@ export class AskarProfileWallet extends AskarBaseWallet {
     try {
       this.walletConfig = walletConfig
 
-      this._session = await this.store.session(walletConfig.id).open()
-
-      // FIXME: opening a session for a profile that does not exist, will not throw an error until
-      // the session is actually used. We can check if the profile exists by doing something with
-      // the session, which will throw a not found error if the profile does not exists,
-      // but that is not very efficient as it needs to be done on every open.
-      // See: https://github.com/hyperledger/aries-askar/issues/163
-      await this._session.fetch({
-        category: 'fetch-to-see-if-profile-exists',
-        name: 'fetch-to-see-if-profile-exists',
-        forUpdate: false,
-        isJson: false,
+      // TODO: what is faster? listProfiles or open and close session?
+      // I think open/close is more scalable (what if profiles is 10.000.000?)
```

Comment:
That would be an interesting stress test! Currently, list_profiles makes a single query (`SELECT name FROM profiles`) and returns all strings at once to FFI layer. In case of millions of records I guess there can be problems in different parts of the stack being used (e.g. maximum variable/array size, SQL query/response length, etc.).

So I think unless there is a new method in askar allowing to check if a particular profile exists, it is safer to do the open/close